commons/src/main/java/org/infinispan/commons/executors/ScheduledThreadPoolExecutorFactory.java

```diff
@@ -13,7 +15,11 @@
 
    @Override
    public ScheduledExecutorService createExecutor(ThreadFactory factory) {
-      ScheduledThreadPoolExecutor result = new ScheduledThreadPoolExecutor(1, factory);
+      ScheduledThreadPoolExecutor result = new ScheduledThreadPoolExecutor(1, factory,
+            // Just use a rejection policy that always throws a lifecycle exception
+            // ScheduledThreadPoolExecutor uses an unbounded queue and always throws a RejectedException for the default
+            // rejection policy.
```

Comment:
Don't need to explain what `AbortPolicy` does, the only surprising part for me is that it's safe to throw `IllegalLifecycleStateException` instead of `RejectedExecutionException` without checking that the executor is stopping.

Reply:
So you are saying to throw `IllegalLifecycleStateException` if it is stopping and `RejectedExecutionException` if it is not? From the code, it will only ever do the former btw. Unless we have a box with so much memory that we hold Integer.MAX_VALUE tasks :)

Reply:
Sorry for pushing so much on this, I know I'm too nitpicky about comments.

I just didn't feel like your comment was explaining why it's ok to always throw an `IllegalLifecycleStateException`, I read it more like a sequence of disconnected facts: Our rejection policy throws this exception, the default policy throws that other exception, and BTW the queue is also unbounded.